### PR TITLE
Add README.md

### LINKED ISSUE #24: A README and description would be nice
I'm learning from this project at Fosdem. It would be nice if this project was easier to find. First steps would be a `README.md` and a project description.

### PR DESCRIPTION
**Pull request by Hoverbear**
This change scaffolds a basic README.md to be added to and improved
later.

Related to #1. Closes #24.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+* [Raft Dissertation](https://github.com/ongardie/dissertation#readme)
+* [Raft Refloated](https://www.cl.cam.ac.uk/~ms705/pub/papers/2015-osr-raft.pdf)
+
+## Using It
+
+Currently `raft-rs` is only available as a Git based dependency. We're working towards making a release available on [crates.io](https://crates.io/) in the future, check out [tracking issue](https://github.com/pingcap/raft-rs/issues/1).
+
+## Developing
+
+`raft` is intended to track the latest nightly for now.
```
Comment (siddontang) [inline README.md]: can support stable version now?

Comment (Hoverbear) [reply README.md]: No, grpcio depends on nightly still. :(

Comment (siddontang) [reply README.md]: @BusyJay can we use stable version for grpio?

Comment (BusyJay) [reply README.md]: I don't get it. What does nightly mean here? grpcio has released several versions and all of them can be compiled with stable rust.

Comment (Hoverbear) [reply README.md]: I was incorrect, it was an unrelated failure. :) It should now work on stable with #8 closed.

**Comment by siddontang** [comment]
LGTM

PTAL @BusyJay

**Comment by siddontang** [comment]
We will add some examples later.

**Comment by BusyJay** [comment]
Travis badges can be added now.

**Comment by siddontang** [comment]
Ping @Hoverbear

**Pull request merged**